modified show courses and displaying info

eleensmathew opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

adding course. Modified models and modified views and html pages for displaying coures.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

Fixes # ([issue](https://github.com/ContriHUB/Youdemy/issues/4))

eleensmathew commented:
Hi @kunal2812  Im facing some difficulty in making an @add_course view because i understand that multiple titles and multiple chapters can be added on the same page. I'm not too sure how to do this. Can you provide some resources how multiple chapters and each chapters titles can be saved in the view?

kunal2812 commented:
You are facing issues in creating schema for the course or saving the entries to the tables ?

eleensmathew commented:
the html page is supposed to save mutliple titles and multiple chapters so im having trouble extracting that information from the html in the view. Can i modify the page so that only one title/one chapter can be added at a time?

kunal2812 commented:
Can you share your schema design ?

eleensmathew commented:
You can see it here https://github.com/ContriHUB/Youdemy/blob/2f1a91de0272e102018cd21ecb7b68264e8f5439/main/models.py

kunal2812 commented:
See, a course can have multiple chapters, so you can create a table called 'Chapters' with columns such as | chapter_id | course_id | content |. Similarly, you can create a table for titles. Whenever you add a new chapter, you'll click on the course link or something to retrieve the course_id

kunal2812 commented:
I'll share some resources for that.

eleensmathew commented:
Yes i have made the tables like that only. I am facing a problem for the add course page where the mentor can choose to add 1 chapter with one title or even 5 chapter with 5 titles each. I am unable to figure out how to send that data to the view and have that data cleaned.

kunal2812 commented:
Check this for now. Will link more.

https://www.youtube.com/watch?v=c2hbT0uIcOQ
https://www.youtube.com/watch?v=PICYTJqj__o
https://www.youtube.com/watch?v=s3T-w2jhDHE

kunal2812 commented:
Check these out

eleensmathew commented:
Hey can I have an extra day? i Have an interview tomorrow so gotta prepare

kunal2812 commented:
Sure. All the best.

eleensmathew commented:
Hi i added all the features but now im debugging the code.

eleensmathew commented:
Hey @kunal2812 th ecode is working. Should I add the option to delete and edit course as well? right now only adding courses ,chapters , titles and publishing courses can be done by mentors.

kunal2812 commented:
Yes you can work on it.

eleensmathew commented:
Hey its done. If you have any comments let me know.

kunal2812 commented:
Great frontend skills, work you've done looks neat.

Has some minor issues will list them for you:

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/faabbab7-68c3-48b4-9526-1719d45c8f0f)

- Info about user not passed to the /courses url so getting user as "None"

- Once the course is published are you not allowing the course mentor to edit it ? It should be editable then too right ? What do you suggest ?

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/1bffe27d-e972-4656-86e9-bfac25177221)

- Fix the navbar on homepage

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/ec853e97-4bff-4a6a-8296-5ab2209749cc)

- Make these more descriptive. For eg. "Course Thumbnail", "Chapter Thumbnail", "Title Thumbnail"

- Is course section removed from the homepage ? I added a course but still can't see it there.

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/c8333d17-3b4d-4eb9-b05e-79cfca260e54)

- Also even though I'm selecting images to upload as thumbnail still I can't seem them (Same as well for chapters)

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/707c3f89-fe87-409a-a567-842568ba2b08)

- You probably missed out this "Questions" part.

eleensmathew commented:
Ok i will work on them as soon as I can
Eleena


On Sat, 21 Oct 2023 at 01:48 Kunal Katiyar ***@***.***> wrote:

> Great frontend skills, work you've done looks neat.
>
> Has some minor issues will list them for you:
>
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277041324-faabbab7-68c3-48b4-9526-1719d45c8f0f.png>
>
>    -
>
>    Info about user not passed to the /courses url so getting user as
>    "None"
>    -
>
>    Once the course is published are you not allowing the course mentor to
>    edit it ? It should be editable then too right ? What do you suggest ?
>
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277041741-1bffe27d-e972-4656-86e9-bfac25177221.png>
>
>    - Fix the navbar on homepage
>
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277042020-ec853e97-4bff-4a6a-8296-5ab2209749cc.png>
>
>    -
>
>    Make these more descriptive. For eg. "Course Thumbnail", "Chapter
>    Thumbnail", "Title Thumbnail"
>    -
>
>    Is course section removed from the homepage ? I added a course but
>    still can't see it there.
>
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277043109-c8333d17-3b4d-4eb9-b05e-79cfca260e54.png>
>
>    - Also even though I'm selecting images to upload as thumbnail still I
>    can't seem them (Same as well for chapters)
>
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277043312-707c3f89-fe87-409a-a567-842568ba2b08.png>
>
>    - You probably missed out this "Questions" part.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1773343101>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HDO6TOSR4BFAOP5W3DYALMABAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZTGM2DGMJQGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

eleensmathew commented:
I think i resolved all of them but unfortunately I'm still unsure why the pictures aren't getting saved? I changed the location from static folder to media but im getting errors maybe because of my migrations? Can you check it out and let me know? @kunal2812

kunal2812 commented:
https://www.youtube.com/watch?v=O5YkEFLXcRg

You can refer to this tutorial to learn to upload images in Django. Also you can look into the Blog feature in the Sheher repository to gain more insight on that.

Also most of the issues mentioned above aren't fixed yet. Nav Bar on home page is still not fixed and image fields aren't made descriptive yet.

eleensmathew commented:
Sure thanks, will fix issues soon
Eleena


On Sun, 22 Oct 2023 at 21:21 Kunal Katiyar ***@***.***> wrote:

> https://www.youtube.com/watch?v=O5YkEFLXcRg
>
> You can refer to this tutorial to learn to upload images in Django. Also
> you can look into the Blog feature in the Sheher repository to gain more
> insight on that.
>
> Also most of the issues mentioned above aren't fixed yet. Nav Bar on home
> page is still not fixed and image fields aren't made descriptive yet.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1774163542>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HE3XFBQLA7EWELCVBLYAVP2VAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZUGE3DGNJUGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

eleensmathew commented:
what exactly should i fix in the nav bar?


On Mon, 23 Oct 2023 at 00:05, Eleena Mathew ***@***.***> wrote:

> Sure thanks, will fix issues soon
> Eleena
>
>
> On Sun, 22 Oct 2023 at 21:21 Kunal Katiyar ***@***.***>
> wrote:
>
>> https://www.youtube.com/watch?v=O5YkEFLXcRg
>>
>> You can refer to this tutorial to learn to upload images in Django. Also
>> you can look into the Blog feature in the Sheher repository to gain more
>> insight on that.
>>
>> Also most of the issues mentioned above aren't fixed yet. Nav Bar on home
>> page is still not fixed and image fields aren't made descriptive yet.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1774163542>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AWAG2HE3XFBQLA7EWELCVBLYAVP2VAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZUGE3DGNJUGI>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

kunal2812 commented:
Frontend is a bit off

How it is now (In Home page and Courses page):
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/2edf05f2-d4f0-4836-be06-655d5cfaf135)

How it should look like:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/ba81737a-8a8e-456e-9bad-44a58f35b2a4)

eleensmathew commented:
I actually added 'Edit your courses' in the nav bar for mentors so they can
access it. Do you want to decrease the padding on the sides or remove the
feature from the navbar all together?

On Mon, 23 Oct 2023 at 00:40, Kunal Katiyar ***@***.***>
wrote:

> Frontend is a bit off
>
> How it is now (In Home page and Courses page):
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277186948-2edf05f2-d4f0-4836-be06-655d5cfaf135.png>
>
> How it should look like:
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277186967-ba81737a-8a8e-456e-9bad-44a58f35b2a4.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1774174897>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HF4X3TOAKSNQQGSDNDYAVVTVAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZUGE3TIOBZG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

eleensmathew commented:
If the issue is uniformity of navbar i have fixed it.

On Mon, 23 Oct 2023 at 02:18, Eleena Mathew ***@***.***> wrote:

> I actually added 'Edit your courses' in the nav bar for mentors so they
> can access it. Do you want to decrease the padding on the sides or remove
> the feature from the navbar all together?
>
> On Mon, 23 Oct 2023 at 00:40, Kunal Katiyar ***@***.***>
> wrote:
>
>> Frontend is a bit off
>>
>> How it is now (In Home page and Courses page):
>> [image: image]
>> <https://user-images.githubusercontent.com/59749539/277186948-2edf05f2-d4f0-4836-be06-655d5cfaf135.png>
>>
>> How it should look like:
>> [image: image]
>> <https://user-images.githubusercontent.com/59749539/277186967-ba81737a-8a8e-456e-9bad-44a58f35b2a4.png>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1774174897>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AWAG2HF4X3TOAKSNQQGSDNDYAVVTVAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZUGE3TIOBZG4>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

kunal2812 commented:
Yes, the problem was with the uniformity of the nav bar.

eleensmathew commented:
![Screenshot 2023-10-23 022746](https://github.com/ContriHUB/Youdemy/assets/92302620/65437363-7a3b-4c49-9282-e8716ee9656f)
Is this okay? Or should i alter the choose file part as well?

kunal2812 commented:
Yes this will work

eleensmathew commented:
I think I fixed all the issues now. Can't believe i'm working on the same issue for a week now. Anyway let me know what you think. If you think I can add something more to get additional points I would love to hear it.

kunal2812 commented:
Course functionality is looking good. 
I see you have added answers to the questions for titles, but are they being stored in a database? And if they are stored, can you make them visible when I click on the question?

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/4420c0d0-f55a-4975-bceb-51503ad53b51)

Nav bar is still not fixed.

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/9dab46a5-ae41-4504-a0f0-0cf0bf6be6f1)

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/e93ee1e0-2b9d-408f-8dee-c8d937d85a89)

This is the problem.

Also, remember that the nav-link was active based on the page we were on. Now, since you have generalized the navbar, this has been removed. It is a minor issue but can be fixed. You can either revert the navbar changes or apply the 'nav-item active' class based on the URL

```
<li class="nav-item active">
       <a href="{% url 'home' %}" class="nav-link">Home</a>
</li>
// Now home remains active on all pages
```

Also, do not remove the courses section from the home page. Courses should still be present on the home page, as they were earlier.

This issue is taking longer than expected. Help me in closing it as soon as possible. Will be adding more issues as soon as this one is resolved so keep an eye out for them.

eleensmathew commented:
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/95bb508c-8f52-44ba-b1eb-56bdcc1bdb75)
this is the updated navbar which i hope is fine? I'm not sure about what you mean by courses in the navbar though.

eleensmathew commented:
I don't think i removed the courses section from the index. i've checked the files changed section so i don't think i've removed a courses section? please let me know what you mean exactly. I have made all the other changes.

kunal2812 commented:
> ![image](https://user-images.githubusercontent.com/92302620/277466027-95bb508c-8f52-44ba-b1eb-56bdcc1bdb75.png) this is the updated navbar which i hope is fine? I'm not sure about what you mean by courses in the navbar though.

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/d92addbb-3698-4ec2-896d-7f352f2bcdb9)

Its still looking like this at my end.

kunal2812 commented:
> I don't think i removed the courses section from the index. i've checked the files changed section so i don't think i've removed a courses section? please let me know what you mean exactly. I have made all the other changes.

Checkout to master branch.

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/651c29f2-c8a9-4a78-99ed-6114e23701e3)

I can see Courses section here.

kunal2812 commented on dbsql:
Remove this from commit.

eleensmathew replied:
done

eleensmathew commented:
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/2046904f-8917-4cfa-8a84-d92ec455f834)
I added it.

kunal2812 commented:
Is navbar fixed ?

eleensmathew commented:
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/b02fadd0-c8af-40b0-8208-ff8bc0bcd1fa)

its looking like this to me so im not sure why its appearing different for you

eleensmathew commented:
i have made changes in styles.css to fix the navbar problem

kunal2812 commented:
Commit all the changes. I'll check again

eleensmathew commented:
I have commited all the changes already.

On Tue, 24 Oct 2023 at 15:32, Kunal Katiyar ***@***.***>
wrote:

> Commit all the changes. I'll check again
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1776902921>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HHNAO7VKXHWMYNGL7TYA6G2TAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZWHEYDEOJSGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kunal2812 commented:
It looks okay when I view it in the Edge browser, but in Chrome, it is not how it should be. Make changes for Chrome as well.

Edge
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/59965e8e-bfac-4024-a8e6-88ba3ee53253)

Chrome
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/e22a63d3-583c-4459-a917-2e630f51d21c)

eleensmathew commented:
I am getting the same display in chrome as well. Can you try clearing the
cache? anyway let me see if i can fix it.

On Tue, 24 Oct 2023 at 15:44, Kunal Katiyar ***@***.***>
wrote:

> It looks okay when I view it in the Edge browser, but in Chrome, it is not
> how it should be. Make changes for Chrome as well.
>
> Edge
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277625801-59965e8e-bfac-4024-a8e6-88ba3ee53253.png>
>
> Chrome
> [image: image]
> <https://user-images.githubusercontent.com/59749539/277625877-e22a63d3-583c-4459-a917-2e630f51d21c.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1776921106>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HAZ7T3OBOLXKWSNNILYA6IGXAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZWHEZDCMJQGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kunal2812 commented:
Yeah. Clearing the cache worked. Reviewing now.

kunal2812 commented:
Courses on "Course" page:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/ce93cee8-e1f5-421e-bf47-e6e519facf54)

Courses on "Home" page:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/435221b1-2f60-470f-8f40-c9cac218d3f4)

Course language should be visible on "Home" page as well.

eleensmathew commented:
> Courses on "Course" page: ![image](https://user-images.githubusercontent.com/59749539/277629987-ce93cee8-e1f5-421e-bf47-e6e519facf54.png)
> 
> Courses on "Home" page: ![image](https://user-images.githubusercontent.com/59749539/277630120-435221b1-2f60-470f-8f40-c9cac218d3f4.png)
> 
> Course language should be visible on "Home" page as well.

fixed

kunal2812 commented:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/4aa691b3-787d-488d-8170-f1456052c2ab)

Fix breadcrumbs in the course page

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/b34e550b-797c-45a4-8c94-f73e8b0726d3)

When I'm on any course "Course" should be highlighted

eleensmathew commented:
Done.

On Tue, 24 Oct 2023 at 16:16, Kunal Katiyar ***@***.***>
wrote:

> [image: image]
> <https://user-images.githubusercontent.com/59749539/277634112-4aa691b3-787d-488d-8170-f1456052c2ab.png>
>
> Fix breadcrumbs in the course page
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1776970781>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HGIVXXKJBMFEGQB2WTYA6MBFAVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZWHE3TANZYGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kunal2812 commented:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/802c85fc-8e1d-4c12-b306-7ce7e87745c9)

No option to add questions when the course has been published.

kunal2812 commented:
Also not able to add chapters. Neither when the course is unpublished nor when it is.

eleensmathew commented:
the current course looks like this 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/151434d3-d34a-4fbe-a50c-2f4dda065aa3)
and 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/5dd3bb95-0d41-44cf-a93b-32645ad920a8)
i have fixed all the highlighted errors. 🫠

kunal2812 commented:
> ![image](https://user-images.githubusercontent.com/59749539/277634582-b34e550b-797c-45a4-8c94-f73e8b0726d3.png)
> 
> When I'm on any course "Course" should be highlighted

This one ?

eleensmathew commented:
Yes fixed that as well 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/0d4d1a14-877a-4f7d-8cf0-a829310d330d)

eleensmathew commented:
ps i deserve 6 billion points

kunal2812 commented:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/2e98e9fa-af0d-47d2-ae81-6d3281eaae1c)

I think title thumbnail field is redundant here. Or is it used somewhere ?

eleensmathew commented:
It is redundant. Do you want me to remove it?
Eleena


On Tue, 24 Oct 2023 at 18:18 Kunal Katiyar ***@***.***> wrote:

> [image: image]
> <https://user-images.githubusercontent.com/59749539/277713777-2e98e9fa-af0d-47d2-ae81-6d3281eaae1c.png>
>
> I think title thumbnail field is redundant here. Or is it used somewhere ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1777463504>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HE6RU6ZMXGOW6H5SCDYA7L55AVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZXGQ3DGNJQGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kunal2812 commented:
Yes

eleensmathew commented:
Done

On Tue, 24 Oct 2023 at 21:05, Kunal Katiyar ***@***.***>
wrote:

> Yes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ContriHUB/Youdemy/pull/12#issuecomment-1777500159>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWAG2HCM4UQYPYQHONQRLNLYA7N47AVCNFSM6AAAAAA6FGPTY2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZXGUYDAMJVHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kunal2812 commented:
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/afb669a1-ab73-40fb-916b-de1f51b76acc)

User with "Student" role is getting these options as well 😶

eleensmathew commented:
Changed that 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/7b867557-790f-41fd-a9cd-6af368702efc)
also added title edit 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/4d9f48aa-5b4e-4970-91a2-0b36e141b82f)

kunal2812 reviewed (approved):
Looking great !!
